addons/hr_holidays/models/hr_leave.py

```diff
@@ -1571,32 +1571,32 @@ def activity_update(self):
         activity_vals = []
         today = fields.Date.today()
         model_id = self.env.ref('hr_holidays.model_hr_leave').id
-        confirm_activity = self.env.ref('hr_holidays.mail_act_leave_approval')
-        approval_activity = self.env.ref('hr_holidays.mail_act_leave_second_approval')
+        confirm_activity = self.env.ref('hr_holidays.mail_act_leave_approval', raise_if_not_found=False)
+        approval_activity = self.env.ref('hr_holidays.mail_act_leave_second_approval', raise_if_not_found=False)
         for holiday in self:
             if holiday.state == 'draft':
-                to_clean |= holiday
+                to_clean += holiday
```

Comment:
Why those changes?

Reply:
Up this too, but I guess it's for performance since we shouldn't care if there's doublets (+ self shouldn't have doublets in an activity)